include/pisa/invert.hpp

```diff
@@ -322,30 +324,26 @@ auto invert_range(gsl::span<gsl::span<Term_Id const>> documents,
     return index;
 }
 
-[[nodiscard]] auto build_batches(std::string const &input_basename,
-                                 std::string const &output_basename,
-                                 uint32_t           term_count,
-                                 size_t             batch_size,
-                                 size_t             threads) -> uint32_t
+[[nodiscard]] auto build_batches(InvertSettings const &settings) -> uint32_t
```

Comment:
I think this should not have `InvertSettings ` as a dependency. I would keep the original signature..

Reply:
Why?

If we don't, the testing does very little in terms of ensuring that we use the right parameters. We can have correct parsing, and correct function, but call it with wrong parameters.